Controllers/CHumidityController.cpp

```diff
@@ -0,0 +1,261 @@
+/*
+ * CHumidityController.cpp
+ *
+ *  Created on: Aug. 11, 2022
+ *      Author: samk
+ */
+
+#include "CHumidityController.h"
+#include <stdio.h>
+#include "CBME280.h"
+#include "ICommand.h"
+#include "IHardwareMap.h"
+
+#define MIN_TEMPERATURE  10
+#define MAX_TEMPERATURE  50
+#define MIN_POWER        0
+#define MAX_POWER        100
+#define DISABLE_OVERRIDE -1
+#define DISABLE_TARGET   0
+#define MIN_HUMIDITY     85
+#define MAX_HUMIDITY     95
+
+CHumidityController::CHumidityController(IHardwareMap *p_hardware,
+                                         etl::string<MAX_STRING_SIZE> name,
+                                         uint32_t run_period_ms,
+                                         SPI_HandleTypeDef *p_spi,
+                                         GPIO_TypeDef *p_slave_select_port,
+                                         uint16_t slave_select_pin)
+    : CController(name, run_period_ms),
+      mp_hw(p_hardware)
+{
+    // TODO Auto-generated constructor stub
+    reset();
+
+    CBME280 hum_temp_reader(p_spi, p_slave_select_port, slave_select_pin);
+    hum_sensor = &hum_temp_reader;
+}
+
+/**
+ * @brief Run all repeated humidity control activities.
+ *
+ */
+void CHumidityController::run()
+{
+    float actual_humidity;
+    for (int i = 0; i < CHANNEL_NUMBER; i++)
+    {
+        float power = 0;
+        actual_humidity = hum_sensor->getHumidity();
```

Comment:
You don't need to call humidity sensor withing the loop. The value of actual humidity is not going to change, so you are just wasting cycles. This entire construct is a bit confusing to me, so perhaps we should have a pair programming session to discuss what you were trying to achieve and how to best approach it.

Reply:
Yes, we can talk about it during the session.